### PR TITLE
virtio_storage_in_use: Add linux support for virtio_storage_in_use

### PR DESCRIPTION
**Pull request by YongxueHong**
1. Support check the backgroud test for linux.
2. Add linux scenario support in virtio_storage_in_use cfg.
3. Add a new case iozone_linux called from virtio_storage_in_use.

ID: 1611971
Signed-off-by: Yongxue Hong yhong@redhat.com

### CODE REVIEW

**Comment by YongxueHong** [comment]
Depends on https://github.com/autotest/tp-qemu/pull/1614

**Comment by YongxueHong** [comment]
@CongLi , @XueqiangWei , @peixiu , @kimi1978wy @lolyu, please help review, thanks.

File: qemu/tests/iozone_linux.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    iozone_cmd_options = params['iozone_cmd_options']
+    iozone_timeout = float(params.get("iozone_timeout", 18000))
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=float(params.get("login_timeout", 360)))
+    error_context.context("Running IOzone command on guest.")
+    iozone = generate_instance(params, session, 'iozone')
+    iozone.run(iozone_cmd_options % iozone_test_dir, iozone_timeout)
+    iozone.clean()
```
Comment (CongLi) [inline qemu/tests/iozone_linux.py]: finally:
iozone.clean()

Comment (YongxueHong) [reply qemu/tests/iozone_linux.py]: Thanks, have updated.

**Comment by YongxueHong** [comment]
@ngu-niny  help me review, thanks.

**Comment by CongLi** [comment]
@YongxueHong 

CI check failed, please help fix it, thanks.

**Comment by hereischen** [comment]
@CongLi  That is because 1614 is not merged yet.

File: qemu/tests/cfg/iozone_linux.cfg [hunk 2]
```diff
@@ -0,0 +1,13 @@
+- iozone_linux: install setup image_copy unattended_install.cdrom
+    virt_test_type = qemu libvirt
```
Comment (hereischen) [inline qemu/tests/cfg/iozone_linux.cfg]: Is it support libvirt?

File: qemu/tests/iozone_linux.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    """
+    Run IOzone for linux on a linux guest:
+    1) Log into a guest.
+    2) Execute the IOzone test.
+
+    :param test: kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    iozone_test_dir = params.get('iozone_test_dir', '/home')
```
Comment (hereischen) [inline qemu/tests/iozone_linux.py]: `iozone_test_dir = params.get('iozone_test_dir', '/home')`
`iozone_test_dir = '/home'`
It is kind of redundant, better make sure it is optional or not.

File: qemu/tests/cfg/iozone_linux.cfg (hunk 2 above)
Comment (YongxueHong) [reply qemu/tests/cfg/iozone_linux.cfg]: thanks, have updated.

File: qemu/tests/iozone_linux.py (hunk 3 above)
Comment (YongxueHong) [reply qemu/tests/iozone_linux.py]: thanks, have updated.

**Comment by ngu-niny** [comment]
Test on virtio_scsi/virtio_blk for a thorough test PASS respectively(on both P8 and P9)

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

**Comment by CongLi** [comment]
Ack.

**Comment by XueqiangWei** [comment]
Tested with below matrix, not hit issue.
guest: linux, windows
drive format: virtio_scsi, virtio_blk
machine type: q35, i440fx

File: qemu/tests/driver_in_use.py [hunk 4]
```diff
@@ -33,6 +33,8 @@ def check_bg_running(target_process):
         list_cmd = params.get("list_cmd", "wmic process get name")
```
Comment (CongLi) [inline qemu/tests/driver_in_use.py]: This command should be windows only.

File: qemu/tests/driver_in_use.py [hunk 5]
```diff
@@ -33,6 +33,8 @@ def check_bg_running(target_process):
...
         output = session.cmd_output_safe(list_cmd, timeout=60)
         process = re.findall(target_process, output, re.M | re.I)
+        if params['os_type'] == 'linux':
+            process = session.cmd_output('pgrep -x %s' % target_process)
```
Comment (CongLi) [inline qemu/tests/driver_in_use.py]: The command does not work for linux guest during migration.

File: qemu/tests/driver_in_use.py (hunk 4 above)
Comment (YongxueHong) [reply qemu/tests/driver_in_use.py]: Thanks, have added judgement for linux and windows

File: qemu/tests/driver_in_use.py (hunk 5 above)
Comment (YongxueHong) [reply qemu/tests/driver_in_use.py]: Thanks, have updated.

**Comment by CongLi** [comment]
Ack.

**Comment by peixiu** [comment]
Tested on win10-32 guest, it works well.

 (01/10) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.system_disk.with_stop_continue.during_bg_test.q35: PASS (604.24 s)
 (02/10) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.system_disk.with_shutdown.during_bg_test.q35: PASS (105.70 s)
 (03/10) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.system_disk.with_reboot.during_bg_test.q35: PASS (456.73 s)
 (04/10) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.system_disk.with_system_reset.during_bg_test.q35: PASS (7577.48 s)
 (05/10) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.system_disk.with_live_migration.during_bg_test.q35: PASS (617.78 s)
 (06/10) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.data_disk.with_stop_continue.during_bg_test.q35: PASS (630.66 s)
 (07/10) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.data_disk.with_shutdown.during_bg_test.q35: PASS (119.87 s)
 (08/10) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.data_disk.with_reboot.during_bg_test.q35: PASS (497.48 s)
 (09/10) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.data_disk.with_system_reset.during_bg_test.q35: PASS (7612.38 s)
 (10/10) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.data_disk.with_live_migration.during_bg_test.q35: PASS (683.74 s)

**Comment by CongLi** [comment]
Ack.

**Comment by peixiu** [comment]
Tested with latest code for rhel8.0.0 guest test, also hit coli upper mentioned issue, 04-Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.system_disk.with_system_reset.during_bg_test.q35t:
test case timeout does not work, case run for a long long time.
After system_reset,  waiting guest to ready run long long time, but I checked the guest status, guest works normally, ip address also normally.follows are waitting guest login messages, I pasted a part of here.

02:01:32 INFO | Context: Run test iozone_linux background --> Reboot guest 'avocado-vt-vm1'. --> rebooting 'avocado-vt-vm1' --> logging in after reboot
02:01:32 DEBUG| Attempting to log into 'avocado-vt-vm1' (timeout 239s)
02:01:32 DEBUG| Found/Verified IP 10.73.75.65 for VM avocado-vt-vm1 NIC 0
02:01:35 DEBUG| STREAM b'IHDR' 16 13
02:01:35 DEBUG| STREAM b'IDAT' 41 1216
02:01:40 DEBUG| STREAM b'IHDR' 16 13
02:01:40 DEBUG| STREAM b'IDAT' 41 1216
02:01:45 DEBUG| STREAM b'IHDR' 16 13
02:01:45 DEBUG| STREAM b'IDAT' 41 1216
02:01:50 DEBUG| STREAM b'IHDR' 16 13
02:01:50 DEBUG| STREAM b'IDAT' 41 1216
02:01:55 DEBUG| STREAM b'IHDR' 16 13
02:01:55 DEBUG| STREAM b'IDAT' 41 1216
02:02:00 DEBUG| STREAM b'IHDR' 16 13
02:02:00 DEBUG| STREAM b'IDAT' 41 1216
02:02:05 DEBUG| STREAM b'IHDR' 16 13
02:02:05 DEBUG| STREAM b'IDAT' 41 1216
02:02:10 DEBUG| STREAM b'IHDR' 16 13
02:02:10 DEBUG| STREAM b'IDAT' 41 1216
02:02:15 DEBUG| STREAM b'IHDR' 16 13
02:02:15 DEBUG| STREAM b'IDAT' 41 1216
02:02:20 DEBUG| STREAM b'IHDR' 16 13
02:02:20 DEBUG| STREAM b'IDAT' 41 1216
02:02:25 DEBUG| STREAM b'IHDR' 16 13
02:02:25 DEBUG| STREAM b'IDAT' 41 1216
02:02:30 DEBUG| STREAM b'IHDR' 16 13
02:02:30 DEBUG| STREAM b'IDAT' 41 1216
02:02:35 DEBUG| STREAM b'IHDR' 16 13
02:02:35 DEBUG| STREAM b'IDAT' 41 1216
02:02:40 DEBUG| STREAM b'IHDR' 16 13
02:02:40 DEBUG| STREAM b'IDAT' 41 1216
02:02:45 DEBUG| STREAM b'IHDR' 16 13
02:02:45 DEBUG| STREAM b'IDAT' 41 1216
02:02:50 DEBUG| STREAM b'IHDR' 16 13
02:02:50 DEBUG| STREAM b'IDAT' 41 1216
02:02:51 DEBUG| Dropped the address cache of HWADDR (9a:ff:00:01:02:03)
02:02:51 DEBUG| Attempting to log into 'avocado-vt-vm1' via serial console (timeout 10s)
02:02:55 DEBUG| STREAM b'IHDR' 16 13
02:02:55 DEBUG| STREAM b'IDAT' 41 1216
02:03:00 DEBUG| STREAM b'IHDR' 16 13
02:03:00 DEBUG| STREAM b'IDAT' 41 1216
02:03:05 DEBUG| STREAM b'IHDR' 16 13
02:03:05 DEBUG| STREAM b'IDAT' 41 1216
02:03:10 DEBUG| STREAM b'IHDR' 16 13
02:03:10 DEBUG| STREAM b'IDAT' 41 1216
02:03:13 WARNI| Error occur when update VM address cache: Login timeout expired    (output: 'exceeded 10 s timeout')
02:03:13 DEBUG| Attempting to log into 'avocado-vt-vm1' via serial console (timeout 10s)
02:03:15 DEBUG| STREAM b'IHDR' 16 13
02:03:15 DEBUG| STREAM b'IDAT' 41 1216
02:03:20 DEBUG| STREAM b'IHDR' 16 13
02:03:20 DEBUG| STREAM b'IDAT' 41 1216
02:03:25 DEBUG| STREAM b'IHDR' 16 13
02:03:25 DEBUG| STREAM b'IDAT' 41 1216
02:03:28 DEBUG| Updated HWADDR (9a:ff:00:01:02:03)<->(10.73.75.65) IP pair into address cache
02:03:30 DEBUG| STREAM b'IHDR' 16 13
02:03:30 DEBUG| STREAM b'IDAT' 41 1216
02:03:35 WARNI| Error occur when update VM address cache: Login timeout expired    (output: 'exceeded 10 s timeout')
02:03:35 DEBUG| STREAM b'IHDR' 16 13
02:03:35 DEBUG| STREAM b'IDAT' 41 1216
02:03:40 DEBUG| STREAM b'IHDR' 16 13
02:03:40 DEBUG| STREAM b'IDAT' 41 1216
02:03:45 DEBUG| STREAM b'IHDR' 16 13
02:03:45 DEBUG| STREAM b'IDAT' 41 1216
02:03:51 DEBUG| STREAM b'IHDR' 16 13
02:03:51 DEBUG| STREAM b'IDAT' 41 1216
02:03:56 DEBUG| STREAM b'IHDR' 16 13
02:03:56 DEBUG| STREAM b'IDAT' 41 1216
02:04:01 DEBUG| STREAM b'IHDR' 16 13
02:04:01 DEBUG| STREAM b'IDAT' 41 1216
02:04:06 DEBUG| STREAM b'IHDR' 16 13
02:04:06 DEBUG| STREAM b'IDAT' 41 1216
02:04:11 DEBUG| STREAM b'IHDR' 16 13
02:04:11 DEBUG| STREAM b'IDAT' 41 1216
02:04:16 DEBUG| STREAM b'IHDR' 16 13
02:04:16 DEBUG| STREAM b'IDAT' 41 1216
02:04:21 DEBUG| STREAM b'IHDR' 16 13
02:04:21 DEBUG| STREAM b'IDAT' 41 1216
02:04:26 DEBUG| STREAM b'IHDR' 16 13
02:04:26 DEBUG| STREAM b'IDAT' 41 1216
02:04:31 DEBUG| STREAM b'IHDR' 16 13
02:04:31 DEBUG| STREAM b'IDAT' 41 1216
02:04:36 DEBUG| STREAM b'IHDR' 16 13
02:04:36 DEBUG| STREAM b'IDAT' 41 1216
02:04:41 DEBUG| STREAM b'IHDR' 16 13
02:04:41 DEBUG| STREAM b'IDAT' 41 1216
02:04:46 DEBUG| STREAM b'IHDR' 16 13
02:04:46 DEBUG| STREAM b'IDAT' 41 1216
02:04:51 DEBUG| STREAM b'IHDR' 16 13
02:04:51 DEBUG| STREAM b'IDAT' 41 1216
02:04:56 DEBUG| STREAM b'IHDR' 16 13
02:04:56 DEBUG| STREAM b'IDAT' 41 1216
02:05:01 DEBUG| STREAM b'IHDR' 16 13
02:05:01 DEBUG| STREAM b'IDAT' 41 1216
02:05:06 DEBUG| STREAM b'IHDR' 16 13
02:05:06 DEBUG| STREAM b'IDAT' 41 1216
02:05:11 DEBUG| STREAM b'IHDR' 16 13
02:05:11 DEBUG| STREAM b'IDAT' 41 1216
02:05:16 DEBUG| STREAM b'IHDR' 16 13
02:05:16 DEBUG| STREAM b'IDAT' 41 1216
02:05:21 DEBUG| STREAM b'IHDR' 16 13
02:05:21 DEBUG| STREAM b'IDAT' 41 1216
02:05:26 DEBUG| STREAM b'IHDR' 16 13
02:05:26 DEBUG| STREAM b'IDAT' 41 1216
02:05:31 DEBUG| STREAM b'IHDR' 16 13
02:05:31 DEBUG| STREAM b'IDAT' 41 1216
02:05:36 DEBUG| STREAM b'IHDR' 16 13
02:05:36 DEBUG| STREAM b'IDAT' 41 1216
02:05:41 DEBUG| STREAM b'IHDR' 16 13
02:05:41 DEBUG| STREAM b'IDAT' 41 1216
02:05:46 DEBUG| STREAM b'IHDR' 16 13
02:05:46 DEBUG| STREAM b'IDAT' 41 1216
02:05:51 DEBUG| STREAM b'IHDR' 16 13
02:05:51 DEBUG| STREAM b'IDAT' 41 1216
02:05:56 DEBUG| STREAM b'IHDR' 16 13
02:05:56 DEBUG| STREAM b'IDAT' 41 1216
02:06:01 DEBUG| STREAM b'IHDR' 16 13
02:06:01 DEBUG| STREAM b'IDAT' 41 1216
02:06:06 DEBUG| STREAM b'IHDR' 16 13
02:06:06 DEBUG| STREAM b'IDAT' 41 1216
02:06:11 DEBUG| STREAM b'IHDR' 16 13
02:06:11 DEBUG| STREAM b'IDAT' 41 1216
02:06:17 DEBUG| STREAM b'IHDR' 16 13
02:06:17 DEBUG| STREAM b'IDAT' 41 1216
02:06:22 DEBUG| STREAM b'IHDR' 16 13
02:06:22 DEBUG| STREAM b'IDAT' 41 1216
02:06:27 DEBUG| STREAM b'IHDR' 16 13
02:06:27 DEBUG| STREAM b'IDAT' 41 1216
02:06:32 DEBUG| STREAM b'IHDR' 16 13
02:06:32 DEBUG| STREAM b'IDAT' 41 1216
02:06:37 DEBUG| STREAM b'IHDR' 16 13
02:06:37 DEBUG| STREAM b'IDAT' 41 1216
02:06:42 DEBUG| STREAM b'IHDR' 16 13
02:06:42 DEBUG| STREAM b'IDAT' 41 1216
02:06:47 DEBUG| STREAM b'IHDR' 16 13
02:06:47 DEBUG| STREAM b'IDAT' 41 1216
02:06:52 DEBUG| STREAM b'IHDR' 16 13
02:06:52 DEBUG| STREAM b'IDAT' 41 1216
02:06:57 DEBUG| STREAM b'IHDR' 16 13
02:06:57 DEBUG| STREAM b'IDAT' 41 1216
02:07:02 DEBUG| STREAM b'IHDR' 16 13
02:07:02 DEBUG| STREAM b'IDAT' 41 1216
02:07:07 DEBUG| STREAM b'IHDR' 16 13
02:07:07 DEBUG| STREAM b'IDAT' 41 1216
02:07:12 DEBUG| STREAM b'IHDR' 16 13
02:07:12 DEBUG| STREAM b'IDAT' 41 1216
02:07:17 DEBUG| STREAM b'IHDR' 16 13
02:07:17 DEBUG| STREAM b'IDAT' 41 1216
02:07:22 DEBUG| STREAM b'IHDR' 16 13
02:07:22 DEBUG| STREAM b'IDAT' 41 1216
02:07:27 DEBUG| STREAM b'IHDR' 16 13
02:07:27 DEBUG| STREAM b'IDAT' 41 1216
02:07:32 DEBUG| STREAM b'IHDR' 16 13
02:07:32 DEBUG| STREAM b'IDAT' 41 1216
02:07:37 DEBUG| STREAM b'IHDR' 16 13
02:07:37 DEBUG| STREAM b'IDAT' 41 1216
02:07:42 DEBUG| STREAM b'IHDR' 16 13
02:07:42 DEBUG| STREAM b'IDAT' 41 1216
02:07:47 DEBUG| STREAM b'IHDR' 16 13
02:07:47 DEBUG| STREAM b'IDAT' 41 1216
02:07:52 DEBUG| STREAM b'IHDR' 16 13
02:07:52 DEBUG| STREAM b'IDAT' 41 1216
02:07:57 DEBUG| STREAM b'IHDR' 16 13
02:07:57 DEBUG| STREAM b'IDAT' 41 1216
02:08:02 DEBUG| STREAM b'IHDR' 16 13
02:08:02 DEBUG| STREAM b'IDAT' 41 1216
02:08:07 DEBUG| STREAM b'IHDR' 16 13
02:08:07 DEBUG| STREAM b'IDAT' 41 1216
02:08:12 DEBUG| STREAM b'IHDR' 16 13
02:08:12 DEBUG| STREAM b'IDAT' 41 1216
02:08:17 DEBUG| STREAM b'IHDR' 16 13
02:08:17 DEBUG| STREAM b'IDAT' 41 1216
02:08:22 DEBUG| STREAM b'IHDR' 16 13
02:08:22 DEBUG| STREAM b'IDAT' 41 1216
02:08:27 DEBUG| STREAM b'IHDR' 16 13
02:08:27 DEBUG| STREAM b'IDAT' 41 1216
02:08:32 DEBUG| STREAM b'IHDR' 16 13
02:08:32 DEBUG| STREAM b'IDAT' 41 1216
02:08:37 DEBUG| STREAM b'IHDR' 16 13
02:08:37 DEBUG| STREAM b'IDAT' 41 1216
02:08:43 DEBUG| STREAM b'IHDR' 16 13
02:08:43 DEBUG| STREAM b'IDAT' 41 1216
02:08:48 DEBUG| STREAM b'IHDR' 16 13
02:08:48 DEBUG| STREAM b'IDAT' 41 1216
02:08:53 DEBUG| STREAM b'IHDR' 16 13
02:08:53 DEBUG| STREAM b'IDAT' 41 1216
02:08:58 DEBUG| STREAM b'IHDR' 16 13
02:08:58 DEBUG| STREAM b'IDAT' 41 1216
02:08:58 DEBUG| Updated HWADDR (f4:8e:38:c3:1d:74)<->(10.73.73.73) IP pair into address cache
02:09:03 DEBUG| STREAM b'IHDR' 16 13
02:09:03 DEBUG| STREAM b'IDAT' 41 1216

..............................................................

02:31:42 DEBUG| STREAM b'IHDR' 16 13
02:31:42 DEBUG| STREAM b'IDAT' 41 1216
02:31:47 DEBUG| STREAM b'IHDR' 16 13
02:31:47 DEBUG| STREAM b'IDAT' 41 1216
02:31:52 DEBUG| STREAM b'IHDR' 16 13
02:31:52 DEBUG| STREAM b'IDAT' 41 1216
.................................

And also saw many errors after guest booted up:
01:59:51 DEBUG| Dropped the address cache of HWADDR (9a:ff:00:01:02:03)
01:59:51 DEBUG| Attempting to log into 'avocado-vt-vm1' via serial console (timeout 10s)
01:59:51 DEBUG| Attempting to log into 'avocado-vt-vm1' via serial console (timeout 10s)
01:59:51 WARNI| Error occur when update VM address cache: Console is in use.
01:59:51 DEBUG| Attempting to log into 'avocado-vt-vm1' via serial console (timeout 10s)
01:59:51 WARNI| Error occur when update VM address cache: Console is in use.
01:59:51 DEBUG| Attempting to log into 'avocado-vt-vm1' via serial console (timeout 10s)
01:59:51 WARNI| Error occur when update VM address cache: Console is in use.
......

File: qemu/tests/iozone_linux.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    iozone_cmd_options = params['iozone_cmd_options']
+    iozone_timeout = float(params.get("iozone_timeout", 18000))
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=float(params.get("login_timeout", 360)))
+    error_context.context("Running IOzone command on guest.")
+    try:
+        iozone = generate_instance(params, session, 'iozone')
+        iozone.run(iozone_cmd_options % iozone_test_dir, iozone_timeout)
```
Comment (peixiu) [inline qemu/tests/iozone_linux.py]: iozone_test_dir will always '/home' here. with_system_disk test, it's ok, but for with_data_disk test,  it should be data disk mounted path, run for /home, that's remain test for system disk.
or you can confirm with xuwei or coli if they want to cover with_data_disk for this case~

Comment (YongxueHong) [reply qemu/tests/iozone_linux.py]: thanks, I have updated,  it's my fault. I forgot it.

File: qemu/tests/iozone_linux.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+                test.error("Failed to get '%s' drive path" % data_image)
+            disks[drive_path[5:]] = data_image
+        return disks
+
+    iozone_test_dir = params.get('iozone_test_dir', '/home')
+    iozone_cmd_options = params['iozone_cmd_options']
+    iozone_timeout = float(params.get("iozone_timeout", 1800))
+    n_partitions = params.get('partitions_num', 1)
+    fstype = params.get('fstype', 'xfs')
+    labeltype = params.get('labeltype', utils_disk.PARTITION_TABLE_TYPE_GPT)
```
Comment (peixiu) [inline qemu/tests/iozone_linux.py]: @XueqiangWei could you confirm which default fstype and labeltype settings for linux basic iozone testing here? for iozone_window, the default remain use NTFS and MBR.

**Comment by peixiu** [comment]
Tried test with latest code on rhel8.0,  5/12 failed with "FAIL: Backgroud test iozone_linux is not alive!".

(01/12) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.q35: PASS (703.66 s)
 (02/12) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.x86_64.io-github-autotest-qemu.rh_kernel_update.q35: PASS (301.95 s)
 (03/12) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.system_disk.with_stop_continue.during_bg_test.q35: FAIL: Backgroud test iozone_linux is not alive! (320.25 s)
 (04/12) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.system_disk.with_shutdown.during_bg_test.q35: PASS (47.51 s)
 (05/12) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.system_disk.with_reboot.during_bg_test.q35: FAIL: Backgroud test iozone_linux is not alive! (314.36 s)
 (06/12) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.system_disk.with_system_reset.during_bg_test.q35: PASS (128.00 s)
 (07/12) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.system_disk.with_live_migration.during_bg_test.q35: FAIL: Backgroud test iozone_linux is not alive! (317.24 s)
 (08/12) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.data_disk.with_stop_continue.during_bg_test.q35: PASS (184.62 s)
 (09/12) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.data_disk.with_shutdown.during_bg_test.q35: FAIL: Backgroud test iozone_linux is not alive! (316.80 s)
 (10/12) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.data_disk.with_reboot.during_bg_test.q35: PASS (118.25 s)
 (11/12) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.data_disk.with_system_reset.during_bg_test.q35: FAIL: Backgroud test iozone_linux is not alive! (316.09 s)
 (12/12) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.0.x86_64.io-github-autotest-qemu.virtio_storage_in_use.with_vioscsi.data_disk.with_live_migration.during_bg_test.q35: PASS (450.26 s)

File: qemu/tests/iozone_linux.py (hunk 7 above)
Comment (XueqiangWei) [reply qemu/tests/iozone_linux.py]: default values are xfs and msdos for linux, but I think xfs and gpt are also ok here. We just focus on virtio storage in use.

**Comment by kimi1978wy** [comment]
Tried with virtio-blk on ws2012r2, no automation errors.

 (01/11) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: PASS (957.11 s)
 (02/11) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest--qemu.virtio_storage_in_use.with_block.system_disk.with_stop_continue.during_bg_test: PASS (132.60 s)
 (03/11) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.virtio_storage_in_use.with_block.system_disk.with_shutdown.during_bg_test: PASS (65.73 s)
 (04/11) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.virtio_storage_in_use.with_block.system_disk.with_reboot.during_bg_test: PASS (148.00 s)
 (05/11) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.virtio_storage_in_use.with_block.system_disk.with_system_reset.during_bg_test: PASS (407.77 s)
 (06/11) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.virtio_storage_in_use.with_block.system_disk.with_live_migration.during_bg_test: PASS (373.34 s)
 (07/11) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.virtio_storage_in_use.with_block.data_disk.with_stop_continue.during_bg_test: PASS (130.00 s)
 (08/11) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.virtio_storage_in_use.with_block.data_disk.with_shutdown.during_bg_test: PASS (73.35 s)
 (09/11) repeat2.Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.virtio_storage_in_use.with_block.data_disk.with_system_reset.during_bg_test: FAIL: iozone test failed: 	Iozone: Performance Test of File I/O\n	        Version $Revision: 3.345 $\n		Compiled for 32 bit mode.\n		Build: Windows \n\n	Contributors:William Norcott, Don Capps, Isom Crawford, Kirby Collins\n	             Al Slater, Scott Rhine, Mike ... (388.51 s)
 (10/11) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.virtio_storage_in_use.with_block.data_disk.with_reboot.during_bg_test: PASS (155.22 s)
 (11/11) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.virtio_storage_in_use.with_block.data_disk.with_live_migration.during_bg_test: PASS (283.12 s)

**Comment by ngu-niny** [comment]
Test on P9 PASS for a whole round test with both virtio_blk and virtio_scsi on both a 7 and 8 guests respectively: 

RESULTS    : PASS 120 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 32831.88 s

Acked-by: Nini Gu <ngu@redhat.com>

File: generic/tests/iozone_windows.py [hunk 8]
```diff
@@ -67,10 +69,16 @@ def get_driver():
...
         error_context.context("Format disk", logging.info)
-        utils_misc.format_windows_disk(session, disk_index,
-                                       mountpoint=disk_letter)
-    cmd = params["iozone_cmd"]
+        formated = not utils_misc.format_windows_disk(session, disk_index,
+                                                      disk_letter)
+        if formated:
+            new_letters = set(utils_misc.get_windows_drive_letters(session))
+            new_letter = list((pre_letters ^ new_letters) & new_letters)
+            # XXX: Replace the disk letter which it maybe changed after vm reboot.
```
Comment (XueqiangWei) [inline generic/tests/iozone_windows.py]: only hit it by system reset, not reboot.

**Comment by CongLi** [comment]
Ack.

**Comment by XueqiangWei** [comment]
tested with virtio-scsi/virio-blk on windows and rhel guest, not hit new issue.

LGTM,
Acked-by xuwei@redhat.com

**Comment by CongLi** [comment]
Ack.

**Comment by YongxueHong** [comment]
Thanks all.

**Pull request merged**